packages/bsp/common/etc/update-motd.d/30-armbian-sysinfo

```diff
@@ -156,7 +156,7 @@ display "Battery" "$battery_percent" "20" "1" "%" "$status_battery_text" ; a=$((
 (( $a > 0 )) && echo "" # new line only if some value is displayed
 
 line=0
-if [[ "$(command -v vnstat)" && $(vnstat --iflist | grep -c "$PRIMARY_INTERFACE") -eq 1 ]]; then
+if vnstat -i "$PRIMARY_INTERFACE" &> /dev/null && [[ -n "$PRIMARY_INTERFACE" ]]; then
```

Comment:
I think you should either remove the `&& [[ -n "$PRIMARY_INTERFACE" ]]` part (since it's not needed), or change the order of the two parts (check first if it's non-empty, _then_ use it)